repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3797,62 +4070,191 @@ public void deleteClassification(String entityGuid, String classificationName) t
+                    put(TASK_CLASSIFICATION_TYPENAME, currentClassification.getTypeName());
+                    put("newMode", true);
+                }};
+
+                taskManagement.createTaskV2(CLASSIFICATION_PROPAGATION_DELETE, currentUser, taskParams, currentClassification.getTypeName(), entityGuid);
+            }
+        }
+
+        // remove classifications from associated entity
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Removing classification: [{}] from: [{}][{}] with edge label: [{}]", classificationName,
+                    getTypeName(entityVertex), entityGuid, CLASSIFICATION_LABEL);
+        }
+
+        tagDAO.deleteDirectTag(entityVertex.getIdForDisplay(), currentClassification);
+
+        RequestContext reqContext = RequestContext.get();
+        // Record cassandra tag operation in RequestContext
+        reqContext.addCassandraTagOperation(entityGuid,
+                new CassandraTagOperation(
+                        entityVertex.getIdForDisplay(),
+                        classificationName,
+                        CassandraTagOperation.OperationType.DELETE,
+                        currentClassification.deepCopy(),
+                        currentTag.getAssetMetadata())
+        );
+
+        List<AtlasClassification> currentTags = tagDAO.getAllClassificationsForVertex(entityVertex.getIdForDisplay());
+
+        Map<String, Map<String, Object>> deNormMap = new HashMap<>();
+        deNormMap.put(entityVertex.getIdForDisplay(), TagDeNormAttributesUtil.getDirectTagAttachmentAttributesForDeleteTag(currentClassification, currentTags, typeRegistry, fullTextMapperV2));
+
+        // ES operation collected to be executed in the end
+        RequestContext.get().addESDeferredOperation(
+                new ESDeferredOperation(
+                        ESDeferredOperation.OperationType.TAG_DENORM_FOR_DELETE_CLASSIFICATIONS,
+                        entityVertex.getIdForDisplay(),
+                        deNormMap
+                )
+        );
+
+        updateModificationMetadata(entityVertex);
+
+        if (RequestContext.get().isDelayTagNotifications()) {
+            RequestContext.get().addDeletedClassificationAndVertices(currentClassification, Collections.singleton(entityVertex));
+        } else {
+            entityChangeNotifier.onClassificationDeletedFromEntities(Collections.singletonList(entityRetriever.toAtlasEntity(entityGuid)), currentClassification);
+        }
+        AtlasPerfTracer.log(perf);
+    }
+
+    private boolean isTaskMatchingWithVertexIdAndEntityGuid(AtlasTask task, String tagTypeName, String entityGuid) {
+        try {
+            if (CLASSIFICATION_PROPAGATION_ADD.equals(task.getType())) {
+                return task.getParameters().get(Constants.TASK_CLASSIFICATION_TYPENAME).equals(tagTypeName)
+                        && task.getParameters().get(ClassificationTask.PARAM_ENTITY_GUID).equals(entityGuid);
+            }
```

Comment:
Isn't this if condition should be based on feature flag? if flag is disabled, we should check on old way